modules/gcp/compute_test.go

```diff
@@ -99,6 +100,105 @@ func TestGetAndSetLabels(t *testing.T) {
 	})
 }
 
+// Set custom metadata on a Compute Instance, and then verify it was set as expected
+func TestGetAndSetMetadata(t *testing.T) {
+	t.Parallel()
+
+	projectID := GetGoogleProjectIDFromEnvVar(t)
+	instanceName := uniqueGcpInstanceName()
+	zone := GetRandomZone(t, projectID, nil, nil)
+
+	// Create a new Compute Instance
+	createComputeInstance(t, projectID, zone, instanceName)
+	defer deleteComputeInstance(t, projectID, zone, instanceName)
+
+	// Set the metadata
+	instance := FetchInstance(t, projectID, instanceName)
+
+	metadataToWrite := map[string]string{
+		"foo": "bar",
+	}
+	instance.SetMetadata(t, metadataToWrite)
+
+	// Now attempt to read the metadata we just set
+	maxRetries := 10
+	sleepBetweenRetries := 3 * time.Second
+
+	retry.DoWithRetry(t, "Read newly set metadata", maxRetries, sleepBetweenRetries, func() (string, error) {
+		instance := FetchInstance(t, projectID, instanceName)
+		metadataFromRead := instance.GetMetadata(t)
+		for _, metadataItem := range metadataFromRead {
+			for key, val := range metadataToWrite {
+				if metadataItem.Key == key && *metadataItem.Value == val {
+					return "", nil
+				}
+			}
+		}
+
+		return "", fmt.Errorf("Metadata that was written was not found in metadata that was read. Retrying.\n")
+	})
+}
+
+// Create a Compute Instance, and attempt to SSH in and run a command.
+func TestSshAccessToComputeInstance(t *testing.T) {
+	t.Parallel()
+
+	projectID := GetGoogleProjectIDFromEnvVar(t)
+	instanceName := uniqueGcpInstanceName()
+	zone := GetRandomZone(t, projectID, nil, nil)
+
+	// Create the Compute Instance
+	createComputeInstance(t, projectID, zone, instanceName)
+	defer deleteComputeInstance(t, projectID, zone, instanceName)
+
+	// Get the public IP
+	var instance *Instance
+	var publicIp string
+
+	maxRetries := 10
+	sleepBetweenRetries := 3 * time.Second
+
+	retry.DoWithRetry(t, "Attempting to get public IP address", maxRetries, sleepBetweenRetries, func() (string, error) {
+		instance = FetchInstance(t, projectID, instanceName)
+
+		publicIp = instance.GetPublicIp(t)
```

Comment:
Could be read via `terraform output` if you use the `examples`/`test` idea mentioned above.